packages/device-utils/src/modelUtils.ts

```diff
@@ -1,3 +1,34 @@
 import { PartialDevice } from './types';
 
-export const getDeviceModel = (device?: PartialDevice) => device?.features?.model || '';
+export enum DeviceModel {
+    T1 = '1',
+    TT = 'T',
+    TR = 'R',
+    UNKNOWN = '',
+}
+
+export const getDeviceModel = (device?: PartialDevice): DeviceModel => {
+    const deviceModel = device?.features?.model;
+
+    if (Object.values(DeviceModel).includes(deviceModel as DeviceModel)) {
+        return deviceModel as DeviceModel;
+    }
+
+    return DeviceModel.UNKNOWN;
+};
+
+export const selectByDeviceModel = (
+    deviceModel: DeviceModel | undefined,
+    options: {
+        default: any;
+        [DeviceModel.T1]?: any;
+        [DeviceModel.TT]?: any;
+        [DeviceModel.TR]?: any;
```

Comment:
try to use `unknown` instead of `any` to keep type checking

Reply:
possibly like this? 
```typescript
export const selectByDeviceModel = <Type>(
    deviceModel: DeviceModel | undefined,
    options: {
        default?: Type;
        [DeviceModel.T1]?: Type;
        [DeviceModel.TT]?: Type;
        [DeviceModel.TR]?: Type;
    },
): Type | undefined => {
    if (!deviceModel || !options[deviceModel]) {
        return options.default;
    }

    return options[deviceModel];
};
```

Then you can have something like
```typescript
const icon = selectByDeviceModel(deviceModel, {
        [DeviceModel.TT]: 'TT',
        [DeviceModel.TR]: 'TR',
    } as const);
```
and `icon` will be type of `'TT' | 'TR'  | undefined` then.

Reply:
4cc84963add469c7441408ed46e6e8fafb6d6171
21f1131aebb38b340d6c3854105d8bb98e302aa1